libsolidity/formal/CHC.cpp

```diff
@@ -382,6 +406,8 @@ bool CHC::visit(WhileStatement const& _while)
 
 bool CHC::visit(ForStatement const& _for)
 {
+	m_scopes.push_back(&_for);
```

Comment:
I am not sure I understand why you need to explicitly do this for the `ForStatement` but not for the `WhileStatement` or the `IfStatement`. Can you explain? The other cases are covered by remembering `Block` scope, right? Why not also this one?

Reply:
Yes, `while` and `if` are not ScopeOpeners, rather their `Block`s are. `ForStatement` is also a `ScopeOpener` because its initialization part may declare new variables that are not inside the block, but neither outside of the `ForStatement`.

Reply:
Makes sense, thanks.